src/Frontend/Components/AttributionList/AttributionList.tsx

```diff
@@ -2,58 +2,139 @@
+import { getManualAttributions } from '../../state/selectors/all-views-resource-selectors';
+import {
+  getMultiSelectSelectedAttributionIds,
+  getSelectedAttributionIdInAttributionView,
+} from '../../state/selectors/attribution-view-resource-selectors';
+import { usePrevious } from '../../util/use-previous';
+import { useFilteredAttributions } from '../../web-workers/use-signals-worker';
+import { Checkbox } from '../Checkbox/Checkbox';
+import { List } from '../List/List';
+import { PACKAGE_CARD_HEIGHT, PackageCard } from '../PackageCard/PackageCard';
+import { SearchTextField } from '../SearchTextField/SearchTextField';
+import { SelectMenu } from '../SelectMenu/SelectMenu';
+import { Spinner } from '../Spinner/Spinner';
+import { ActionBar, ButtonGroup, Container } from './AttributionList.style';
+import {
+  getFilteredAndSortedPackageCardIdsAndDisplayPackageInfos,
+  useFilterMenuOptions,
+} from './AttributionList.util';
 
 export function AttributionList() {
   const dispatch = useAppDispatch();
   const selectedPackageCardIdInAttributionView = useAppSelector(
     getSelectedAttributionIdInAttributionView,
   );
+  const multiSelectSelectedAttributionIds = useAppSelector(
+    getMultiSelectSelectedAttributionIds,
+  );
+  const selectedAttributionIdAttributionView = useAppSelector(
+    getSelectedAttributionIdInAttributionView,
+  );
+  const manualAttributions = useAppSelector(getManualAttributions);
+
+  const [anchorEl, setAnchorEl] = useState<HTMLElement>();
 
-  const { attributions, activeFilters } = useFilteredAttributions();
-  const hasActiveFilters = !!activeFilters.length;
-  const [showMultiSelect, setShowMultiselect] = useState(hasActiveFilters);
+  const [{ attributions, loading, search }, setFilteredAttributions] =
+    useFilteredAttributions();
+  const { filteredAndSortedIds, filteredAndSortedAttributions } = useMemo(
+    () =>
+      getFilteredAndSortedPackageCardIdsAndDisplayPackageInfos(
+        attributions,
+        search,
+      ),
+    [attributions, search],
+  );
+
+  const effectiveSelectedIds = useMemo(
+    () => intersection(filteredAndSortedIds, multiSelectSelectedAttributionIds),
+    [filteredAndSortedIds, multiSelectSelectedAttributionIds],
+  );
+  const prevEffectiveSelectedIds = usePrevious(
+    effectiveSelectedIds,
+    effectiveSelectedIds,
+  );
+
+  useEffect(() => {
+    if (!isEqual(effectiveSelectedIds, prevEffectiveSelectedIds)) {
+      dispatch(setMultiSelectSelectedAttributionIds(effectiveSelectedIds));
+    }
+  }, [dispatch, effectiveSelectedIds, prevEffectiveSelectedIds]);
+
+  const { selectedFilters, options: filterMenuOptions } =
+    useFilterMenuOptions();
+  const someSelectedAttributionsArePreSelected = useMemo(
+    () =>
+      multiSelectSelectedAttributionIds.some(
+        (id) => filteredAndSortedAttributions[id]?.preSelected,
+      ),
+    [filteredAndSortedAttributions, multiSelectSelectedAttributionIds],
+  );
+
+  const areAllAttributionsSelected = useMemo(
```

Comment:
Variable names for booleans should not be phrased as questions.
```suggestion
  const allAttributionsAreSelected = useMemo(
```

Reply:
it's common practice in many programming languages to start booleans with "is". this PR is not the right place to start such a fundamental discussion.